src/useTimer.test.tsx

```diff
@@ -61,6 +61,37 @@ describe('Start', () => {
     expect(getByTestId('time').textContent).toBe('15');
   });
 
+  it('should start timer on updated initialTime value', () => {
+    // Given
+    const Component = ({ initialTime }: { initialTime: number }) => {
+      const { time, start } = useTimer({
+        initialTime,
+      });
+
+      return (
+        <div>
+          <button onClick={start}>Start</button>
+          <p data-testid="time">{time}</p>
+        </div>
+      );
+    };
+
+    const { getByRole, getByTestId, rerender } = render(
+      <Component initialTime={10} />
+    );
+    rerender(<Component initialTime={20} />);
+
+    // When
+    fireEvent.click(getByRole('button'));
+
+    act(() => {
+      jest.advanceTimersByTime(5000);
+    });
+
+    // Then
+    expect(getByTestId('time').textContent).toBe('25');
+  });
```

Comment:
I think that this test is not complete. Here, You just start the timer only once, after the second rendering. 
If I understood well, according to [your example](https://stackblitz.com/edit/react-pecbfx?file=src/App.js), what you want to test is:
- Start timer with a first initial value.
- Change this initial value and re-render component.
- Reset timer.
- Start timer and check that new initial value is correctly displayed.

You can check my suggestion. I just added some little changes to your test.

```suggestion
  it('should re-start timer with updated initialTime', () => {
    // Given
    const Component: React.FC<Partial<Config>> = ({ initialTime }) => {
      const { time, start, reset } = useTimer({
        initialTime,
      });

      return (
        <div>
          <button data-testid="start" onClick={start}>
            Start
          </button>
          <button data-testid="reset" onClick={reset}>
            Reset
          </button>
          <p data-testid="time">{time}</p>
        </div>
      );
    };

    const { getByTestId, rerender } = render(<Component initialTime={10} />);
    const startButton = getByTestId('start');
    const resetButton = getByTestId('reset');

    // When
    fireEvent.click(startButton);

    rerender(<Component initialTime={20} />);

    // When
    fireEvent.click(resetButton);
    fireEvent.click(startButton);

    // Then
    expect(getByTestId('time').textContent).toBe('20');
  });
```